pallets/investments/src/mock.rs

```diff
@@ -134,18 +135,30 @@ impl pallet_investments::Config for Runtime {
 	type CollectedRedemptionHook = NoopCollectHook;
 	type InvestmentId = InvestmentId;
 	type MaxOutstandingCollects = MaxOutstandingCollect;
-	type PreConditions = Always;
+	type PreConditions = AlwaysWithOneException;
 	type RuntimeEvent = RuntimeEvent;
 	type Tokens = OrmlTokens;
 	type WeightInfo = ();
 }
 
-pub struct Always;
-impl<T> PreConditions<T> for Always {
+pub struct AlwaysWithOneException;
+impl<T> PreConditions<T> for AlwaysWithOneException
+where
+	T: Into<OrderType<AccountId, InvestmentId, Balance>> + Clone,
+{
 	type Result = DispatchResult;
 
-	fn check(_: T) -> Self::Result {
-		Ok(())
+	fn check(order: T) -> Self::Result {
+		match order.clone().into() {
+			OrderType::Investment { who, .. } | OrderType::Redemption { who, .. }
+				if who == NOT_INVESTOR =>
+			{
+				Err(DispatchError::Other(
+					"PreCondition mock fails on u64::MAX account on purpose",
+				))
+			}
+			_ => Ok(()),
+		}
```

Comment:
NOTE: I did not want to refactor UTs here to solely use the mock-builder. We should tackle this tech debt later this year because I don't think `pallet-investments` has great coverage. Until this point, preconditions were completely irrelevant. The provided solution is only temporary. Please let me know if you disagree with the proposal.

Reply:
Agree, `pallet-investment` needs anyways a cleaning over the mock and tests. A lot has been learned from when it was developed